metricshub-web/react/src/components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,507 @@
+import * as React from "react";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { Box, Drawer, Typography, useMediaQuery } from "@mui/material";
+
+// LocalStorage key for persistent drawer width
+const LS_KEY = "metricshub.drawerWidth";
+
+// Drawer width bounds (px)
+const DEFAULT_WIDTH = 320;
+const MIN_WIDTH = 200;
+const MAX_WIDTH = 1000;
+
+// Overlay scrollbar geometry (px)
+const THUMB_WIDTH = 8;
+const TRACK_INSET = 0;
+const MIN_THUMB = 28;
+
+// Sash UX tuning
+const SASH_HOVER_DELAY = 500;
+const SASH_MOVE_THRESHOLD = 4;
+const SASH_KEY_STEP = 10;
+
+/**
+ * Sidebar (tree-agnostic)
+ *
+ * Props:
+ *  - open: boolean
+ *  - onOpenChange: (open: boolean) => void
+ *  - onWidthChange?: (width: number) => void  // reported on desktop
+ *  - appBarHeights?: { xs: number, sm: number }
+ *  - activeTree?: string                       // which tree key to render
+ *  - trees?: Record<string, React.ReactNode>   // registry of trees (optional)
+ *  - emptyState?: React.ReactNode              // optional custom empty-state node
+ */
+export default function Sidebar({
+	open,
+	onOpenChange,
+	onWidthChange,
+	appBarHeights = { xs: 56, sm: 64 },
+	activeTree = "none",
```

Comment:
This component is supposed to accept any component on either left or right side.

Left side can be a treeview, but it could be something else as well! (think of another splitter, horizontal, for several sections, like in VSCode).

If possible, let's keep this as generic as possible.

Example of structure of the UI:

```html
<MetricsHub UI>
  <NavBar />
  <MainContent route="console?resource=localhost">
    <Console resource="localhost">
      <SplitScreen>
        <Left>
          <ConsoleTree resource="localhost" />
        </Left>
        <Right>
          <ConsoleContent resource="localhost" />
        </Right>
      </SplitScreen>
    </Console>
  </MainContent>
</MetricsHub UI>
```